### PR TITLE
When specifying only a filename for miDebuggerPath, search $Path for debugger

### PR DESCRIPTION
**Pull request by pieandcakes**
[vscode-cpptools Issue 442](https://github.com/Microsoft/vscode-cpptools/issues/442)
@gregg-miskelly @chuckries @rajkumar42

### CODE REVIEW

File: src/MICore/LaunchOptions.cs [hunk 1]
```diff
@@ -330,6 +328,20 @@ static internal LocalLaunchOptions CreateFromXml(Xml.LaunchOptions.LocalLaunchOp
                     throw new InvalidLaunchOptionsException(MICoreResources.Error_NoMiDebuggerPath);
                 }
             }
+            else
+            {
+                // If user specifies only a filename for miDebuggerPath, search the local PATH to see if we can determine where it is. 
+                if (!File.Exists(miDebuggerPath) &&
+                    miDebuggerPath.IndexOfAny(new char[] { Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar }) == 0)
+                {
```
Comment (rajkumar42) [inline src/MICore/LaunchOptions.cs]: -1 ? 
https://msdn.microsoft.com/en-us/library/11w09h50(v=vs.110).aspx

**Comment by gregg-miskelly** [comment]
Can you update the description in LaunchOptions.xsd as well?

File: src/MICore/LaunchOptions.cs (hunk 1 above)
Comment (pieandcakes) [inline src/MICore/LaunchOptions.cs]: Fixed

File: src/MICore/LaunchOptions.xsd [hunk 2]
```diff
@@ -225,7 +225,7 @@
           </xs:sequence>  
           <xs:attribute name="MIDebuggerPath" type="xs:string" use="required">
             <xs:annotation>
-              <xs:documentation>Full path to the MI Debugger executable.</xs:documentation>
+              <xs:documentation>Path to the MI Debugger executable. If only the executable name is provided, it will attempt to search $PATH for the location of the executable.</xs:documentation>
```
Comment (pieandcakes) [inline src/MICore/LaunchOptions.xsd]: @gregg-miskelly Is this the change you are referring to?

Comment (gregg-miskelly) [inline src/MICore/LaunchOptions.xsd]: Yes

**Pull request merged**